File: silbot3_tutorials/src/device/control_led.cpp
#include "ros/ros.h"
#include "silbot3_msgs/Device_LED_Msg.h"

ros::Publisher publisher;

void on(int r, int g, int b)
{
  silbot3_msgs::Device_LED_Msg msg;
  msg.command = "on";
  msg.id = 0xF5;
  msg.bright = 1;
  msg.red = r;
  msg.green = g;
  msg.blue = b;
  publisher.publish(msg);
  ROS_INFO("published");
}

void controlLed() {
  on(1,1,1);
  ros::Duration(1).sleep();
  on(1,0,0);
  ros::Duration(1).sleep();
  on(0,1,0);
  ros::Duration(1).sleep();
  on(0,0,1);
  ros::Duration(1).sleep();
  on(0,0,0);
}

int main(int argc, char **argv)
{
  ros::init(argc, argv, "control_led_cpp");
  ros::NodeHandle n;
  publisher = n.advertise<silbot3_msgs::Device_LED_Msg>("/DeviceNode/LED/commands", 1000);

  int count = 0;
  while (count < 0) {
    count = publisher.getNumSubscribers();
    ROS_INFO("wait device led subscriber...");
    ros::Duration(0.1).sleep();
  }
  ROS_INFO("found device led subscriber...");
  ros::Duration(0.5).sleep();
  controlLed();

  return 0;
}

File: scripts/device/control_led.py
#! /usr/bin/env python
import rospy
from silbot3_msgs.msg import  Device_LED_Msg

class LED_Control :
    def __init__(self):
        self.publisher = rospy.Publisher("/DeviceNode/LED/commands", Device_LED_Msg, queue_size=10)

        while self.publisher.get_num_connections() < 1 :
            rospy.loginfo("Wait LED subscriber")
            rospy.sleep(0.1)

    def on(self, r, g, b):
        msg = Device_LED_Msg()
        msg.command = "on"
        msg.id = 0xF5
        msg.bright = 1
        msg.red = r
        msg.green = g
        msg.blue = b

        self.publisher.publish(msg)
        rospy.loginfo("published")


if __name__ == "__main__" :
    rospy.init_node("control_led_python", anonymous=False)
    led = LED_Control()
    led.on(1, 1, 1)
    rospy.sleep(1)
    led.on(1, 0, 0)
    rospy.sleep(1)
    led.on(0, 1, 0)
    rospy.sleep(1)
    led.on(0, 0, 1)
    rospy.sleep(1)
    led.on(0, 0, 0)
